packages/ui/v2/modules/booker/DatePicker.tsx

```diff
@@ -215,8 +245,15 @@ const DatePicker = ({
           </div>
         ))}
       </div>
-      <div className="grid grid-cols-7 gap-1 text-center">
-        <Days weekStart={weekStart} selected={selected} {...passThroughProps} browsingDate={browsingDate} />
+      <div className="relative grid grid-cols-7 gap-1 text-center">
+        <Days
+          weekStart={weekStart}
+          selected={selected}
+          {...passThroughProps}
+          browsingDate={browsingDate}
+          month={month}
+          nextMonthButton={() => changeMonth(+1)}
```

Comment:
we need these two props for the overlay. preferably we move the component out of `<Days>` and do all the calculation on the backend to speed up the frontend.

it should be:

`{isNotAvailable ? <NoAvailabilityOverlay/> : <Days/>}`

Reply:
On the other hand this would make this UI fairly difficult, as both Days and the overlay are being drawn.
<img width="470" alt="image" src="https://user-images.githubusercontent.com/1046695/207729812-9201b4aa-14d6-4f43-9647-203d36c3e1fa.png">